src/components/cl/basic/cl_basic_lib.c

```diff
@@ -31,10 +33,19 @@ UCC_CLASS_DEFINE(ucc_cl_basic_lib_t, ucc_cl_lib_t);
 ucc_status_t ucc_cl_basic_get_lib_attr(const ucc_base_lib_t *lib,
                                        ucc_base_attr_t *base_attr)
 {
-    ucc_cl_lib_attr_t *attr      = ucc_derived_of(base_attr, ucc_cl_lib_attr_t);
-    attr->tls                    = UCC_TL_UCP | UCC_TL_NCCL;
-    attr->super.attr.thread_mode = UCC_THREAD_SINGLE;
-    /* TODO: fill coll_types, reduction_types, sync_mode.
-       Correctly fill thead_mode multiple when possible. */
+    ucc_cl_lib_attr_t *attr           = ucc_derived_of(base_attr, ucc_cl_lib_attr_t);
+    attr->tls                         = UCC_TL_UCP | UCC_TL_NCCL;
+    ucc_component_iface_t *ucp_iface  =
+    		ucc_get_component(&ucc_global_config.tl_framework, "ucp");
+    ucc_tl_iface_t *tl_ucp_iface      = ucc_derived_of(ucp_iface, ucc_tl_iface_t);
+    attr->super.attr.thread_mode      = tl_ucp_iface->max_thread_level;
```

Comment:
why do you directly use iface field instead of calling tl get lib attr?

Reply:
there is no TL/LIB to query at this point ..

Reply:
same true for cl, isn't it? There is no cl basic lib